### PR TITLE
Remove QRCodeGeneratorView in favor of data: URIs

### PR DESCRIPTION
**Pull request by akx**
Extracted from #69 with some additional fixes.

### CODE REVIEW

File: allauth_2fa/middleware.py [hunk 1]
```diff
@@ -95,6 +96,15 @@ def process_view(self, request, view_func, view_args, view_kwargs):
...
+        for urlname in self.allowed_pages:
+            try:
+                if request.path == reverse(urlname):
+                    return
+            except NoReverseMatch:
+                # The developer may have misconfigured the list of allowed pages.
+                # Let's not outright crash at that point, but inform the developer about their mishap.
+                warnings.warn('NoReverseMatch for %s while checking for pages allowed without 2FA' % urlname)
+
         if request.path in map(reverse, self.allowed_pages):
```
Comment (clokep) [inline allauth_2fa/middleware.py]: I believe this code is now duplicative to the newly added for-loop.

File: allauth_2fa/utils.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
+from base64 import b32encode
+
+from django.contrib.sites.shortcuts import get_current_site
+from django.utils.six import BytesIO
+
+import qrcode
+from qrcode.image.svg import SvgPathImage
```
Comment (clokep) [inline allauth_2fa/utils.py:11]: Does this require any additional packages to be installed? (I assume not, but just checking!)

File: allauth_2fa/middleware.py (hunk 1 above)
Comment (akx) [reply allauth_2fa/middleware.py]: My bad.

File: allauth_2fa/utils.py (hunk 2 above)
Comment (akx) [reply allauth_2fa/utils.py:11]: Nope!

It's what the original code uses anyway.

https://github.com/percipient/django-allauth-2fa/blob/a63008ff79049942fcafcc72a856ee13722ea35e/allauth_2fa/views.py#L24

**Comment by akx** [comment]
The test failure seems to be some sort of SSL shenanigan between Bitbucket and Travis. 😐

**Comment by clokep** [comment]
Going to try it again and see what happens. 😄

**Review by clokep — APPROVED** [review]
Looks good. I rebased this after fixing master.

**Pull request merged**